docs/examples/basic-price-oracle/main.go

```diff
@@ -0,0 +1,163 @@
+// This example demonstrates a basic RPC price oracle server that implements the
+// QueryRateTick RPC method. The server listens on localhost:8095 and returns a
+// rate tick for a given transaction type, subject asset, and payment asset. The
+// rate tick is the exchange rate between the subject asset and the payment
+// asset.
+package main
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"time"
+
+	oraclerpc "github.com/lightninglabs/taproot-assets/taprpc/priceoraclerpc"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+)
+
+const (
+	// serviceListenAddress is the listening address of the service.
+	serviceListenAddress = "localhost:8095"
+)
+
+// RpcPriceOracleServer is a basic example RPC price oracle server.
+type RpcPriceOracleServer struct {
+	oraclerpc.UnimplementedPriceOracleServer
+}
+
+// isSupportedSubjectAsset returns true if the given subject asset is supported
+// by the price oracle, and false otherwise.
+func isSupportedSubjectAsset(subjectAsset *oraclerpc.AssetSpecifier) bool {
+	// Ensure that the subject asset is set.
+	if subjectAsset == nil {
+		return false
+	}
+
+	// In this example we'll only support a single asset.
+	assetIdStr := subjectAsset.GetAssetIdStr()
+	supportedAssetId := "7b4336d33b019df9438e586f83c587ca00fa65602497b9" +
+		"3ace193e9ce53b1a67"
+
+	return assetIdStr == supportedAssetId
+}
+
+// getRateTick returns a rate tick for a given transaction type and subject
+// asset max amount.
+func getRateTick(transactionType oraclerpc.TransactionType,
+	subjectAssetMaxAmount uint64) oraclerpc.RateTick {
+
+	// Determine the rate based on the transaction type.
+	var rate uint64
+	if transactionType == oraclerpc.TransactionType_PURCHASE {
```

Comment:
Can we add more godoc details to the diff of the two transaction types? Perhaps with a brief walkthrough example? So something like: Alice has `X` and wants to pay an invoice, she queries the oracle for a quote with `transactionType` `Z`. 

On second thought, this sounds like something we should add as normal mark down docs, and also in the builder's guide.

Reply:
In the latest commits I've included an example in the doc for `QueryRateTick`.